object_store/src/aws/mod.rs

```diff
@@ -605,4 +606,71 @@ mod tests {
             store.delete(location).await.unwrap();
         }
     }
+
+    /// See CONTRIBUTING.md for the MinIO setup for this test.
+    #[tokio::test]
+    async fn test_s3_ssec_encryption_with_minio() {
+        if std::env::var("TEST_S3_SSEC_ENCRYPTION").is_err() {
+            eprintln!("Skipping S3 SSE-C encryption test");
+            return;
+        }
+        eprintln!("Running S3 SSE-C encryption test");
+
+        let customer_key = "1234567890abcdef1234567890abcdef";
+        let expected_md5 = "JMwgiexXqwuPqIPjYFmIZQ==";
+
+        let store = AmazonS3Builder::new()
+            .with_bucket_name("test-bucket")
+            .with_access_key_id("minio")
+            .with_secret_access_key("minio123")
+            .with_ssec_encryption(customer_key)
+            .with_endpoint("https://localhost:9000")
```

Comment:
Instead of hard coding these values in the binary, could we take them from environment variables? That way we can just change the environment variables to point them at an AWS S3 bucket to re-use the test.
```suggestion
        let store = AmazonS3Builder::from_env()
            .with_ssec_encryption(customer_key)
```